plugins/backend/catalog-backend-module-okta/README.md

```diff
@@ -5,17 +5,104 @@ entity providers to read Okta Group and User objects as Backstage Entities.
 
 To setup the Okta providers you will need an [Okta API Token](https://developer.okta.com/docs/guides/create-an-api-token/main/)
 
-You will need to configure the providers in your catalog.ts file in your backstage backend:
+## App Config
 
-## OktaUserEntityProvider
+You will need to configure your okta credentials in the `app-config.yaml`.
+
+### Basic Config
+
+```yaml
+catalog:
+  providers:
+    okta:
+      - orgUrl: 'https://tenant.okta.com'
+        token:
+          $env: OKTA_TOKEN
+```
+
+### Filter Users and Groups
+
+```yaml
+catalog:
+  providers:
+    okta:
+      - orgUrl: 'https://tenant.okta.com'
+        token:
+          $env: OKTA_TOKEN
+        userFilters:
+          - key: profile.department
+            operator: equals
+            value: engineering
+        groupFilters:
+          - key: profile.name
+            operator: startsWith
+            value: engineering_
+```
+
+There are two ways that you can configure the Entity providers. You can either use the `OktaOrgEntityProvider` which loads both users and groups. Or you can load user or groups separately user the `OktaUserEntityProvider` and `OktaGroupEntityProvider` providers.
+
+## Load Users and Groups Together
+
+### OktaOrgEntityProvider
 
 You can configure the provider with different naming strategies. The configured strategy will be used to generate the discovered entity's `metadata.name` field. The currently supported strategies are the following:
 
+User naming stategies:
+
 - id (default) | User entities will be named by the user id.
 - kebab-case-email | User entities will be named by their profile email converted to kebab case.
 - strip-domain-email | User entities will be named by their profile email without the domain part.
 
-## OktaGroupEntityProvider
+Group naming strategies:
+
+- id (default) | Group entities will be named by the group id.
+- kebab-case-name | Group entities will be named by their group profile name converted to kebab case.
+
+### Example configuration:
+
+```typescript
+import { OktaOrgEntityProvider } from '@roadiehq/catalog-backend-module-okta';
+
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
+  const builder = await CatalogBuilder.create(env);
+
+  const oktaConfig = env.config.getOptionalConfigArray(
+    'catalog.providers.okta',
+  );
```

Comment:
I think that this piece of logic should be the responsibility of the provider's static fromConfig method, otherwise it assumes it gets a Config object which has okta specific keys on it.

Reply:
Done